### DISCUSSION TITLE
Can we add the original content to the style configuration?

### DISCUSSION THREAD

**Discussion by noobnooc** [gonzalezreal/swift-markdown-ui#201] [Ideas]
I would like to add a copy button to the code block, but I am unable to find a method to do so. Below, you may find the code that I have attempted thus far:

```swift
Markdown(content.trimmingCharacters(in: .whitespacesAndNewlines))
            .markdownBlockStyle(\.codeBlock) { configuration in
                ScrollView(.horizontal) {
                    configuration.label
                        .padding(10)
                }
                .markdownTextStyle(textStyle: {
                    FontFamilyVariant(.monospaced)
                    FontSize(.em(0.85))
                })
                .background(Color.primary.opacity(0.1))
                .cornerRadius(8)
                .padding(.bottom)
                .textSelection(.enabled)

                // TODO: - Get the content
                .overlay(alignment: .bottomTrailing) {
                    Button {
                        // Here is the coping mechanism, but it seems like there is no way to retrive the code content.
                        // Can we add the code content to the configuration?
                    } label: {
                        Image(systemName: "doc.on.doc")
                    }
                    .tint(.secondary)
                    .frame(height: 25)
                    .buttonStyle(.borderedProminent)
                    .padding(10)
                }
            }
```

Maybe we can add the code content to the configuration? Or is there another way to do this?

**Comment by gonzalezreal** [comment]
Hi @noobnooc,

Have you considered using SwiftUI's text selection API? By applying the `textSelection(_:)` modifier to the `Markdown` view, you can allow users to copy the contents.

```swift
Markdown {
  """
  You can quote text with a `>`.
  > Outside of a dog, a book is man's best friend. Inside of a
  > dog it's too dark to read.
  – Groucho Marx
  """
}
.textSelection(.enabled)
```

While it would be possible to add a `CodeBlockConfiguration` type that provides the code block text content to allow implementing a copy button, the styling API is intended only for appearance customization. I am open to reconsidering this if more use cases require accessing the text contents from the block-style body.

**Comment by mmohades** [comment]
+1 on this. It'd be nice to have the option of modifying blocks, like the code-blcok.

**Comment by mmohades** [comment]
Also, text selection doesn't seem to work on my simulator. :/
```
[Pasteboard] ...saving pasteboard failed with error: Error Domain=NSItemProviderErrorDomain Code=-1000 "Cannot load representation of type com.apple.uikit.attributedstring" UserInfo={NSLocalizedDescription=Cannot load representation of type com.apple.uikit.attributedstring, NSUnderlyingError=0x6000007caf10 {Error Domain=NSCocoaErrorDomain Code=4866 "Caught exception during archival: -[__NSCFType encodeWithCoder:]: unrecognized selector sent to instance 0x600002cae880
```

Edit: works fine on the phone

**Comment by graddotdev** [comment]
@gonzalezreal, I added the copy button, but then I realized that I don't have access to the code. That's really unfortunate.

<img width="478" alt="Screenshot 2023-03-29 at 17 10 58" src="https://user-images.githubusercontent.com/4836709/228547351-59c72092-ae78-4a7e-81ae-8fe87a37d342.png">

**Comment by gonzalezreal** [comment]
I am working on something to enable this functionality. Unfortunately, it can take a while because I need to refactor a few things internally. In the meantime, please use the `textSelection(_:)` modifier as a workaround.

**Reply by graddotdev** [reply]
Currently, I am using my own implementation of markdown rendering, but yours looks much better. It's a lot of work, so thank you for that. I will be waiting for the new feature.

**Reply by gonzalezreal** [reply]
FYI: relevant to this discussion, #215 and #220 have been merged into the `main` branch and should fulfill this feature request.

**Comment by wscqs** [comment]
![e4a0646982dad13525f560138b0fde5e](https://user-images.githubusercontent.com/13110076/233814864-651d932c-39b6-485a-bc85-6a8c9445d6c1.png)
need copybutton   +1

**Reply by noobnooc** [reply]
The latest version `2.1.0` has the capacity to implement a copy button.

**Reply by kmmao** [reply]
I updated to version 2.1.0, but I'm not sure how to properly call the copy button.

**Reply by moophis** [reply]
You can refer to the code https://github.com/gonzalezreal/swift-markdown-ui/pull/249 for the copy functionality.

**Comment by wscqs** [comment]
来信已经收到。

**Comment by wscqs** [comment]
来信已经收到。
